### PR TITLE
Log warnings when registration/node API used

### PR DESCRIPTION
**Pull request by azdagron**
The APIs will be deprecated in a future release. This change logs a deprecation warning once per RPC per hour.

Further, a light migration guide for migrating Registration API clients is provided.

Fixes: #1994

### CODE REVIEW

File: doc/migration_registration_api_clients.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Migrating Registration API Clients
+
+The `registration` API has been deprecated and will be removed from a future
+release. The new API surface is a superset of the previous functionality and
+provides consistent semantics, batch, and paging support.
+
+This document outlines the replacement RPCs when migrating clients away from
+the old registration API.
+
+## Replacement RPCs
```
Comment (amartinezfayo) [inline doc/migration_registration_api_clients.md:10]: The `ListFederatedBundles` RPC is missing here.

**Review by amartinezfayo — COMMENTED** [review]
Thank you for this @azdagron!

File: doc/migration_registration_api_clients.md (hunk 1 above)
Comment (azdagron) [reply doc/migration_registration_api_clients.md:10]: Whoa! I must have accidentally deleted the line in my typing fury! Thanks for catching that.

**Review by amartinezfayo — APPROVED** [review]
🚀

**Pull request merged**